vyper/parser/self_call.py

```diff
@@ -81,6 +82,11 @@ def call_self_private(stmt_expr, context, sig):
     pop_return_values = []
     push_args = []
 
+    if len(stmt_expr.keywords):
```

Comment:
I don't think self public calls will correctly handle any keyword arguments either so I think this restriction should also apply to self public calls.  This check should probably be moved to `self_call.call_lookup_specs` or `self_call.make_call`.  Is my thinking correct here @jacqueswww ?

Reply:
Tests should then also be added for invalid self public calls with keyword args.

Reply:
See #1558 - I originally implemented this on both public and private calls, but now I'm working on #1199 which will make the change to public calls irrelevant.  So modified my PR to only cover private calls for now.  If it is preferred to also block them in public calls for now, happy to reopen the previous one or modify this one.